frontend/rolecall/src/app/piece/piece_editor.component.ts

```diff
@@ -1,108 +1,117 @@
-import { CdkDragDrop, transferArrayItem } from '@angular/cdk/drag-drop';
-import { Location } from '@angular/common';
-import { Component, OnInit } from '@angular/core';
-import { MatSelectChange } from '@angular/material/select';
-import { ActivatedRoute } from '@angular/router';
-import { COLORS } from 'src/constants';
-import { isNullOrUndefined } from 'util';
-import { Piece, PieceType, PieceApi, Position } from '../api/piece_api.service';
-import { ResponseStatusHandlerService } from '../services/response-status-handler.service';
-import { APITypes } from 'src/api_types';
+import {CdkDragDrop, transferArrayItem} from '@angular/cdk/drag-drop';
+import {Location} from '@angular/common';
+import {Component, OnInit} from '@angular/core';
+import {MatSelectChange} from '@angular/material/select';
+import {ActivatedRoute} from '@angular/router';
+import {COLORS} from 'src/constants';
+import {Piece, PieceType, PieceApi, Position} from '../api/piece_api.service';
+import {ResponseStatusHandlerService} from '../services/response-status-handler.service';
+import {APITypes} from 'src/api_types';
 
 
 type DraggablePosition = {
-  index: number,
-  value: Position,
-  valueName: "New Position" | "Existing Position" | "New Ballet" | "Existing Ballet",
-  type: "adding" | "added" | "editing",
-  nameDisplay: string,
-  sizeDisplay: string,
-}
+  index: number;
+  pos: Position;
```

Comment:
We're defining a new name `pos` to replace the `value` property here, right? It's already way better, but would be nice if it was the full word `position`, which would also line up nicely with some other usages later in this file.

Reply:
I struggled with this one. The problem is that the 'Position' object is inside another object of type 'DraggablePosition'. And worse, the outer objects are almost always called xxxPositions. I worried that calling the inside object position too would make people think it was of type 'DraggablePosition', like all the other positions. I thought if I called it pos, it would look different enough to give some indication that it was different.

If we call this object position, we probably should change the names of all enclosing objects.

Reply:
Ok let's leave this can of worms closed for now.